pkg/cmd/factory/default.go

```diff
@@ -14,11 +16,93 @@ import (
 )
 
 func New(appVersion string) *cmdutil.Factory {
-	io := iostreams.System()
+	f := &cmdutil.Factory{
+		Config:     configFunc(), // No factory dependencies
+		Branch:     branchFunc(), // No factory dependencies
+		Executable: executable(), // No factory dependencies
+	}
+
+	f.IOStreams = ioStreams(f)                   // Depends on Config
+	f.HttpClient = httpClientFunc(f, appVersion) // Depends on Config, IOStreams, and appVersion
+	f.Remotes = remotesFunc(f)                   // Depends on Config
+	f.BaseRepo = BaseRepoFunc(f)                 // Depends on Remotes
+	f.Browser = browser(f)                       // Depends on IOStreams
+
+	return f
+}
+
+func BaseRepoFunc(f *cmdutil.Factory) func() (ghrepo.Interface, error) {
+	return func() (ghrepo.Interface, error) {
+		remotes, err := f.Remotes()
+		if err != nil {
+			return nil, err
+		}
+		return remotes[0], nil
+	}
+}
+
+func SmartBaseRepoFunc(f *cmdutil.Factory) func() (ghrepo.Interface, error) {
```

Comment:
Formerly `resolvedBaseRepo` I moved this to be near the other `BaseRepo` function.